lib/trino-collect/src/main/java/io/trino/collect/cache/NonKeyEvictableCacheImpl.java

```diff
@@ -49,4 +52,46 @@ public void invalidateAll(Iterable<?> keys)
         throw new UnsupportedOperationException("invalidateAll(keys) does not invalidate ongoing loads, so a stale value may remain in the cache for ever. " +
                 "Use EvictableCache if you need invalidation");
     }
+
+    @Override
+    public ConcurrentMap<K, V> asMap()
+    {
+        ConcurrentMap<K, V> map = delegate.asMap();
+        return new ForwardingConcurrentMap<K, V>()
+        {
+            @Override
+            protected ConcurrentMap<K, V> delegate()
+            {
+                return map;
+            }
+
+            @Override
+            public V remove(Object key)
+            {
+                throw new UnsupportedOperationException("remove(key) does not invalidate ongoing loads, so a stale value may remain in the cache for ever. " +
+                        "Use EvictableCacheBuilder if you need invalidation");
+            }
+
+            @Override
+            public boolean remove(Object key, Object value)
+            {
+                throw new UnsupportedOperationException("remove(key, value) does not invalidate ongoing loads, so a stale value may remain in the cache for ever. " +
+                        "Use EvictableCacheBuilder if you need invalidation");
+            }
+
+            @Override
+            public boolean replace(K key, V oldValue, V newValue)
```

Comment:
This one may actually be safe? 🤔

Reply:
If we forbid without being sure, we will appear as if being sure.

I am fine with forbidding this, but let's have a comment that we don't know what we're doing.